pkg/config/node.go

```diff
@@ -226,52 +226,55 @@ func GetConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVip
 	node.Cluster.VIPNetmask = prefix
 	node.VRRPInterface = vipIface.Name
 
-	domain := fmt.Sprintf("%s.%s", clusterName, clusterDomain)
-	node.LBConfig, err = GetLBConfig(domain, apiPort, lbPort, statPort, apiVip)
+	return node, err
+}
+
+func getSortedBackends(kubeconfigPath string) (backends []Backend, err error) {
+	var masterIp string
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
 	if err != nil {
-		return node, err
+		log.WithFields(logrus.Fields{
+			"err": err,
+		}).Info("Failed to get client config")
+		return []Backend{}, err
 	}
-	etcdBackends, err := getSortedBackends(domain)
+	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		return node, err
-	}
-	for _, backend := range etcdBackends {
-		if len(node.Cluster.EtcdBackends) > 0 {
-			node.Cluster.EtcdBackends += ","
-		}
-		node.Cluster.EtcdBackends += fmt.Sprintf("https://%s:2379", strings.TrimRight(backend.Host, "."))
+		log.WithFields(logrus.Fields{
+			"err": err,
+		}).Info("Failed to get client")
+		return []Backend{}, err
 	}
-
-	return node, err
-}
-
-func getSortedBackends(domain string) (backends []Backend, err error) {
-	srvs, err := utils.GetEtcdSRVMembers(domain)
+	nodes, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/master=",
+	})
 	if err != nil {
 		log.WithFields(logrus.Fields{
 			"err": err,
-		}).Info("Failed to get Etcd SRV members")
-		srvs = []*net.SRV{}
-		err = nil
+		}).Info("Failed to get master Nodes list")
+		return []Backend{}, err
 	}
-
-	for _, srv := range srvs {
-		addr, err := utils.GetFirstAddr(srv.Target)
-		if err != nil {
-			log.WithFields(logrus.Fields{
-				"member": srv.Target,
-			}).Error("Failed to get address for member")
-			continue
+	for _, node := range nodes.Items {
+		masterIp = ""
+		for _, address := range node.Status.Addresses {
+			if address.Type == v1.NodeInternalIP {
+				masterIp = address.Address
+				break
+			}
+		}
+		if masterIp != "" {
```

Comment:
Should we log if this check fails? Seems like something would be wrong if that's the case.

Reply:
Agreed

Reply:
Done